vortex-array/src/arrays/varbin/vtable/operator.rs

```diff
@@ -0,0 +1,184 @@
+    fn bind(
+        array: &VarBinArray,
+        selection: Option<&ArrayRef>,
+        ctx: &mut dyn BindCtx,
+    ) -> VortexResult<BatchKernelRef> {
+        let mask = ctx.bind_selection(array.len(), selection)?;
+        let validity = ctx.bind_validity(array.validity(), array.len(), selection)?;
+        let offsets = ctx.bind(array.offsets(), None)?;
+
+        match array.dtype() {
+            DType::Utf8(_) => Ok(Box::new(VarBinKernel::<StringType>::new(
+                offsets,
+                array.bytes().clone(),
+                validity,
+                mask,
+            ))),
+            DType::Binary(_) => Ok(Box::new(VarBinKernel::<BinaryType>::new(
+                offsets,
+                array.bytes().clone(),
+                validity,
+                mask,
+            ))),
+            _ => unreachable!("invalid DType for VarBinArray {}", array.dtype()),
+        }
+    }
+}
+
+struct VarBinKernel<V> {
+    offsets: BatchKernelRef,
+    bytes: ByteBuffer,
+    validity: MaskExecution,
+    selection: MaskExecution,
+    _type: PhantomData<V>,
+}
+
+impl<V> VarBinKernel<V> {
+    fn new(
+        offsets: BatchKernelRef,
+        bytes: ByteBuffer,
+        validity: MaskExecution,
+        selection: MaskExecution,
+    ) -> Self {
+        Self {
+            offsets,
+            bytes,
+            validity,
+            selection,
+            _type: PhantomData,
+        }
+    }
+}
+
+impl<V: BinaryViewType> BatchKernel for VarBinKernel<V> {
+    fn execute(self: Box<Self>) -> VortexResult<Vector> {
+        let offsets = self.offsets.execute()?.into_primitive();
+
+        match_each_integer_ptype!(offsets.ptype(), |T| {
+            let pvec = offsets.downcast::<T>();
+            // NOTE: discard the validity because offsets must be non-nullable
+            let (offsets, _) = pvec.into_parts();
+            let first = offsets[0];
+
+            let lens: Buffer<u32> = offsets
+                .iter()
+                .copied()
+                .skip(1)
+                .scan(first, |prev, next| {
+                    let len = (next - *prev)
+                        .to_u32()
+                        .vortex_expect("offset must map to u32");
+                    *prev = next;
+                    Some(len)
+                })
+                .collect();
+
+            let mut views = BufferMut::with_capacity(lens.len());
+
+            for (offset, len) in std::iter::zip(offsets, lens) {
+                let offset = offset.to_u32().vortex_expect("offset must fit in u32");
+                let bytes = &self.bytes[offset as usize..(offset + len) as usize];
+                let view = if len as usize <= BinaryView::MAX_INLINED_SIZE {
+                    BinaryView::new_inlined(bytes)
+                } else {
+                    BinaryView::make_view(bytes, 0, offset)
+                };
+                views.push(view);
+            }
+
+            let selection = self.selection.execute()?;
```

Comment:
Feels like we should do this first? Then do like a `selection.threshold_iter()` to iterate over indices and slices when constructing the views?